### PR TITLE
feat: Add `contents_as_json` column to `github_workflows` table

### PR DESCRIPTION
**Pull request by akash1810**
#### Summary

Adds a new column `contents_as_json` to the `github_workflows` table, representing the Workflow's content in JSON form.

The `contents` column of this table is stringified YAML. Postgres doesn't have first class support for querying YAML; JSON columns can be natively queried.

The column's resolver gets the value in the `contents` column, and uses `github.com/ghodss/yaml` to convert it to JSON. This is similar to how [AWS CloudFormation templates](https://github.com/cloudquery/cloudquery/blob/fb65286719f2924f3be7fcc8d45b01a95fa09c97/plugins/source/aws/resources/services/cloudformation/stack_templates.go) are processed.

❓ I'm not sure if the mechanism used to get the `contents` column is preferable. It is different from [the docs](https://docs.cloudquery.io/docs/developers/creating-new-plugin/go-source#writing-a-column-resolver) because we've different types - the table has `github.Workflow` and the column `github.RepositoryContent`. Is this likely to cause issues?

### CODE REVIEW

**Review by erezrokah — CHANGES_REQUESTED** [review]
Hi @akash1810  and thanks for the PR.
I understand the use case of querying JSON instead of YAML but there are a few things to consider:
1. A workflow file can be invalid YAML either because of a syntax error, or simply because GitHub workflows YAML syntax differs from the spec, so we can't assume the content is YAML

2. The PR to fix [AWS CloudFormation templates](https://github.com/cloudquery/cloudquery/pull/10826) is a different use case. We assumed the data to be JSON, however later discovered it can be YAML, so we try to coerce the data to be JSON, and added a new column with the raw data, for the JSON column to be deprecated sometime in the future.

Generally CloudQuery doesn't do any transformation on the data and saves it as raw as possible, since there can be endless use cases on how to transform it. We found it's better to do any transformations post sync, and I believe you could run a query to add this column if needed

Please let me know if that makes sense

**Comment by erezrokah** [comment]
Maybe opening an issue with your specific use case will help clarify the need for the column? Could be easier to discuss this over an issue

File: plugins/source/github/resources/services/actions/workflows.go [hunk 1]
```diff
@@ -86,3 +94,17 @@ func resolveContents(ctx context.Context, meta schema.ClientMeta, resource *sche
 	}
 	return resource.Set(c.Name, content)
 }
+
+func resolveContentsAsJson(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
```
[the remark this answers is not in the document]
Comment (akash1810) [reply plugins/source/github/resources/services/actions/workflows.go:98]: Ah! That's a lot cleaner!

**Comment by akash1810** [comment]
> Maybe opening an issue with your specific use case will help clarify the need for the column? Could be easier to discuss this over an issue

Issue opened - https://github.com/cloudquery/cloudquery/issues/16850

**Comment by erezrokah** [comment]
Hi @akash1810, sorry for the delay. We discussed this change internally and agreed it makes sense to be able to query the workflow YAML via JSON syntax as suggested in the issue https://github.com/cloudquery/cloudquery/issues/16850.

We've considered recommending using a Postgres extension to add a virtual column/view to handle the YAML conversion, however realized that might not work for hosted Postgres solutions.

Would you like to follow up on this PR with my suggestion from https://github.com/cloudquery/cloudquery/pull/16846#discussion_r1501055216 to extend the existing resolver?

**Comment by akash1810** [comment]
> We discussed this change internally and agreed it makes sense to be able to query the workflow YAML via JSON syntax as suggested in the issue https://github.com/cloudquery/cloudquery/issues/16850.

Ah excellent.

> Would you like to follow up on this PR with my suggestion from https://github.com/cloudquery/cloudquery/pull/16846#discussion_r1501055216 to extend the existing resolver?

Will do! I'll hope to get this done this week.

Any thoughts on schema validation (as mentioned in https://github.com/cloudquery/cloudquery/issues/16850) here? If CloudQuery tries to represent the source APIs as closely as possible, its probably out of scope to perform a schema check here too?

**Comment by erezrokah** [comment]
> its probably out of scope to perform a schema check here too?

Is the purpose to rely on the schema to query the YAML? If so I think we could verify it.
However the file could still be valid YAML without matching the schema (in case of user error), so you'd lose querying capabilities in that case. Another issue is that the schema can change over time so you'd need to know which schema was used during the sync to validate.

I'd say not validate for now, and write queries that rely on the schema but can gracefully handle invalid one if not too complicated

**Comment by akash1810** [comment]
> I'll hope to get this done this week.

Pushed an update now. I've not written much Go, so any feedback on code style, etc welcomed!

**Review by erezrokah — APPROVED** [review]
@akash1810 thanks for the following, just got around to testing this. Works great 🚀 
Nice one with the no-op resolver

**Comment by erezrokah** [comment]
OK added https://github.com/cloudquery/cloudquery/pull/16846/commits/7b9006747a3de346d3d74abec720290c48bdafa8 as I don't think we should fail if YAML parsing fail. We should still keep the text content. If that makes sense to you I'll go ahead and merge the PR

**Comment by akash1810** [comment]
> OK added [7b90067](https://github.com/cloudquery/cloudquery/commit/7b9006747a3de346d3d74abec720290c48bdafa8) as I don't think we should fail if YAML parsing fail. We should still keep the text content. If that makes sense to you I'll go ahead and merge the PR

LGTM!

Thanks also for the `style:` commit; I forgot to run `make lint` before pushing...

**Pull request merged**